polygeist/test/polygeist-opt/sycl/licm-accessor-versioning.mlir

```diff
@@ -111,6 +107,94 @@ func.func private @test(%arg0: memref<?x!sycl_accessor_1_i32_rw_gb, 4>, %arg1: m
 
 // -----
 
+!sycl_id_1_ = !sycl.id<[1], (!sycl.array<[1], (memref<1xi64>)>)>
+!sycl_range_1_ = !sycl.range<[1], (!sycl.array<[1], (memref<1xi64>)>)>
+!sycl_accessor_impl_device_1_ = !sycl.accessor_impl_device<[1], (!sycl_id_1_, !sycl_range_1_, !sycl_range_1_)>
+!sycl_accessor_1_i32_rw_gb = !sycl.accessor<[1, i32, read_write, global_buffer], (!sycl_accessor_impl_device_1_, !llvm.struct<(memref<?xi32, 1>)>)>
+!sycl_accessor_1_i32_r_gb = !sycl.accessor<[1, i32, read, global_buffer], (!sycl_accessor_impl_device_1_, !llvm.struct<(memref<?xi32, 1>)>)>
+
+// CHECK: [[GUARD_COND:#.*]] = affine_set<() : (7 >= 0)>
+
+// COM: There is only one loop, i.e., the loop is not versioned.
+// CHECK-LABEL: testAffineLoopVersioning
```

Comment:
Should we simplify this test like the test below, and rely on the first test to test the codegen?

Reply:
Sure